runtime/src/bank.rs

```diff
@@ -7691,6 +7691,46 @@ impl Bank {
             .epoch_accounts_hash_manager
             .try_get_epoch_accounts_hash()
     }
+
+    /// Write details about this Bank's stored accounts to a file
+    pub fn write_stored_accounts_to_file_for_debugging(&self) {
+        // We're not equipped to compare results on a non-frozen bank
+        if !self.is_frozen() {
+            return;
+        }
+        let slot = self.slot();
+
+        let file_name = format!("{}.{}", slot, self.hash());
+        let path = self
+            .rc
+            .accounts
+            .accounts_db
+            .get_accounts_hash_cache_path()
+            .join(&file_name);
+        if path.exists() {
+            return;
+        }
+
+        let mut accounts = self
+            .rc
+            .accounts
+            .accounts_db
+            .get_stored_accounts_for_slot(slot);
+        // Account pubkeys are the first item in the summary, so we can
+        // just sort based on the entire string for easy diffing
+        accounts.sort();
+
+        use std::fs;
+        // Log the last blockhash incase we have a bad version of the block
```

Comment:
Just to make sure you talking about the wrong bank hash or the bad version of a block comment that is shown in this comment?
- If wrong bank hash, that makes sense although I think it might be more appropriate to add that increment to ReplayStage
- If wrong blockhash, we don't have the information available to determine if blockhash caused the failing bank hash.

Reply:
I'm going to defer adding the metric in this PR, but echoing my above statement, I think we should add it to similar spot that https://github.com/solana-labs/solana/pull/28596 added panic in ReplayStage.

That being said, the above mentioned PR has a panic after so many failures, to that provides some signaling. But, I guess a metric could be useful too (ie someone could fail once and recover, which could indicate some non-determinism)